spring-cloud-app-broker-deployer-cloudfoundry/src/main/java/org/springframework/cloud/appbroker/deployer/cloudfoundry/CloudFoundryAppDeployer.java

```diff
@@ -565,23 +627,25 @@ private String getLastUpdatedPackageId(List<PackageResource> packageResources) {
 				.memory(memory(properties))
 				.putAllEnvironmentJsons(environmentVariables)
 				.build())
-			.doOnRequest(l -> LOG.debug("Updating environment for application {}", applicationId))
-			.doOnSuccess(response -> LOG.info("Updated environment for application {}", applicationId))
-			.doOnError(e -> LOG.warn(
-				String.format("Error Updating environment for application %s: %s", applicationId, e.getMessage())));
+			.doOnRequest(l -> LOG.debug("Updating environment. applicationId={}", applicationId))
+			.doOnSuccess(response -> {
+				LOG.info("Success updating environment. applicationId={}", applicationId);
+				LOG.debug(RESPONSE_LOG_TEMPLATE, response);
+			})
+			.doOnError(e -> LOG.warn(String.format("Error updating environment. applicationId=%s, " +
+				ERROR_LOG_TEMPLATE, applicationId, e.getMessage()), e));
 	}
 
 	private Function<Flux<Long>, Publisher<?>> getExponentialBackOff() {
 		return DelayUtils.exponentialBackOff(Duration.ofSeconds(2), Duration.ofMinutes(5), Duration.ofMinutes(10));
 	}
 
-	private Mono<Void> pushApplication(DeployApplicationRequest request,
-		Map<String, String> deploymentProperties,
+	private Mono<Void> pushApplication(DeployApplicationRequest request, Map<String, String> deploymentProperties,
 		Resource appResource) {
 		ApplicationManifest manifest = buildAppManifest(request, deploymentProperties, appResource);
 
 		if (LOG.isDebugEnabled()) {
-			LOG.debug("Pushing manifest" + manifest.toString());
+			LOG.debug("Pushing app manifest. manifest={}" + manifest.toString());
```

Comment:
We've got a format string combined with string concatenation here 😄  It'll log an extra pair of `{}` - `... manifest={}<manifest contents>`

Reply:
oops